File: online-judge-solutions/Codeforces/1634A.py
#https://codeforces.com/problemset/problem/1634/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	4400 KB
t=int(input())
for x in range(0,t):
    n,k=map(int,input().split())
    s=input()
    if(k==0):print(1)
    else:
        if(s[::-1]==s):print(1)
        else:print(2)

File: online-judge-solutions/Codeforces/1634A.cpp
//https://codeforces.com/problemset/problem/1634/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	1000 KB
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
	int t,n,k;
	cin>>t;
	while(t--){
		string s;
		cin>>n>>k>>s;
		string r=s;
		reverse(s.begin(),s.end());
		if(k==0) cout<<"1\n";
		else{
			if(s==r){cout<<"1\n";}
			else{cout<<"2\n";}
		}
	}
	return 0;
}
